### DISCUSSION TITLE
Turning off S2 Lolin mini blue standby led

### DISCUSSION THREAD

**Discussion by dmytro-frolov** [awawa-dev/HyperSerialESP32#53] [Announcements]
Hi,

Setup:
I have *ESP32-S2 Lolin mini* and *apa107/hd107* LED strip, the power goes from a 5v external power supply to buffer *SN74AHCT125N* connected to a smart socket. 

Issue:
Unfortunately, when I power off the power supply the *s2 mini* still ignites the *blue standby led* and some LEDs from the strip, I think the *SN74AHCT125N* passes some current to the strip. 

My main question is, can I turn off *s2 mini* when I don't need it, using API or something? When I connect ESP to USB, the *standby led* and LEDs don't spark until I turn on the LED device under Remote Control on the dashboard

**Comment by awawa-dev** [comment]
Hi
Could you draw a simple diagram of how you connected it all? I suspect there is something wrong with the configuration (or SN74AHCT125N is damaged) because this is the first time I have heard of such a problem. If by some miracle it passes the data lines to the 5v line, would the esp32-s2 GPIO power the entire LED strip through the level shifter (this would overload the esp32 instantly)? If there is really a shortcut, +5V in the other direction also would kill the esp32 GPIO instantly.

**Comment by dmytro-frolov** [comment]
Thank you for the replay,

Sure, tried to attach the diagram. Keep in mind that nothing is overloaded or overheated. If I turn off the power the ESP powers few LEDs. Those LEDs are red and dim. Until the *blue standby led* is on no matter what I do within hyperion nothing changes LEDs behaviour. If I plug out the USB and plug in back, the *standby blue led* is off as well as the LED strip.

I have 10A 5V power supply and 134 LEDs on the hd107 strip around the TV frame. 

<img width="920" alt="image" src="https://github.com/awawa-dev/HyperSerialESP32/assets/1667831/f4c58f64-5319-41f3-b245-d19a1ae46673">

**Comment by awawa-dev** [comment]
Still this behavior is not correct and I cannot explain it from an electrical point of view how this is possible. Something's missing here. However, to me this situation looks very dangerous. Apart from the lack of power (have you checked the 5V lines with a multimeter?), the addressable LEDs will not light up without a command. Sometimes noise can trigger them, but this is rather unusual for SPI, as such an event should occur simultaneously on both CLOCK and DATA lines. Single-line Neopixels are much more sensitive and sometimes the fluctuations during startup (especially on GPIO2 when UART is present on esp32 or esp8266) can turn on some diodes, provided they have power.

I haven't used ESP32-s2 mini for a long time, so it may be a dead end, but I think the blue LED is connected to GPIO15 in this model... does this LED turn on again after disconnecting/connecting ESP to USB and restarting the LED driver in HyperHDR? How is the unit to which the ESP is connected powered?

**Comment by awawa-dev** [comment]
> . If I turn off the power the ESP powers few LEDs. 

If you disconnect jack (LED power supply on the diagram) not esp, are they still lit?

I'm moving the subject to the discussion.

**Comment by dmytro-frolov** [comment]
>If you disconnect jack (LED power supply on the diagram) not esp, are they still lit?

Yes, a few of them, they are really dim and red

>does this LED turn on again after disconnecting/connecting ESP to USB and restarting the LED driver in HyperHDR? How is the unit to which the ESP is connected powered?

I hope this can bring some light


Power adapter off:
1. Connect USB
- No LEDs, no blue standby led
2. Hyperion LED device **ON** (Screenshot)
- Blue led is on, LED strips are red. Just noticed, that leds that are at the beginning of the strip blink in the response to grabber, but they are just red. I think they are just red because the colour red consumes less power than other colours. (photo.png the red arrow pointing to blue standby led)
3. Hyperion LED device **OFF** (Screenshot)
- The blue led still on, strip is red, but in the static. I think there weren't reset data or it was inefficient
1.63V between GND and VCC on the strip.
4. Hyperion LED device **ON**
- The blue led still on, strip is red. Still static as from the previous step. 
1.63V between GND and VCC on the strip.
5. When I turn off the power, everything works as expected (Photo2)


>I think the blue LED is connected to GPIO15 

When does the blue light start to glow? I tried to connect it to USB and sending some text directly to the device. I'm trying to understand when clock and data output is generated, in my opinion they give parassite current to the LED strip


<img width="450" alt="image" title="screenshot" src="https://github.com/awawa-dev/HyperSerialESP32/assets/1667831/fbaf0c22-0665-444c-8558-cc0657e577b2">
<img width="450" alt="image" title="Photo 1" src="https://github.com/awawa-dev/HyperSerialESP32/assets/1667831/bbe35b62-eabf-4f25-93eb-bc748f91b544">
<img width="450" alt="image" title="Photo 2" src="https://github.com/awawa-dev/HyperSerialESP32/assets/1667831/0675e432-c88d-49fe-8536-f1d8692e77e9">

**Comment by dmytro-frolov** [comment]
Hello @awawa-dev,

After a while, I discovered that the blue LED is responsible for the active spi connection to the LED strip, 

In this draft PR https://github.com/dmytro-frolov/HyperSerialESP32/pull/1/files I close SPI connection to LED once UART connection from the host closed, it's not anything like production-ready solution, but please look at it. 

It solves voltage drop and the LEDs don't ignite on their own, unfortunately, handshake doesn't work, but I've been using this version for more than a week and everything seems to be fine, and I'm going to stick to it
